ArduCopter/heli.cpp

```diff
@@ -180,4 +180,34 @@ void Copter::heli_update_rotor_speed_targets()
     rotor_runup_complete_last = motors->rotor_runup_complete();
 }
 
+#if MODE_AUTOROTATE_ENABLED == ENABLED
+// heli_update_autorotation - determines if aircraft is in autorotation and sets motors flag and switches
+// to autorotation flight mode if manual collective is not being used.
+void Copter::heli_update_autorotation()
+{
```

Comment:
Add code here that enables/disables the spdhgt enable parameter based on arot->is_enabled().  you may have to add accessors in the spdhgt control library to allow you to do this.

Reply:
I have had a go at implementing this.  I have tested it and generally it works.  The only problem I run into is when setting AROT_ENABLE = 0 and refreshing the parameters, it tends to hang whilst fetching the parameters list.  'Rebooting' the sim sorts the problem and everything works fine again.  I'm not sure if this is acceptable 'reboot required' behaviour or whether the way I have done this isn't the right way of doing it.